### PR TITLE
Fix nightly builds on master

### PR DESCRIPTION
**Pull request by omichel**
Fixes the creation of a non-draft prerelease for nightly build.

### CODE REVIEW

File: scripts/packaging/publish_release.py [hunk 1]
```diff
@@ -120,9 +120,7 @@
                         except requests.exceptions.ConnectionError:
                             remainingTrials -= 1
                             print('Release upload failed (remaining trials: %d)' % remainingTrials)
-                    if (releaseExists and
-                            not (options.tag and not options.tag.startswith('refs/heads/')) and
-                            not releaseCommentModified and
+                    if (releaseExists and tag.startswith('nightly_') and not releaseCommentModified and
```
Comment (lukicdarkoo) [inline scripts/packaging/publish_release.py:123]:
```suggestion
                    if (releaseExists and not release.draft and not releaseCommentModified and
```
I am not completely sure whether it is going to work. But I believe it is better to define a logic (whether it is a draft or not) in the single place.

File: scripts/packaging/publish_release.py [hunk 2]
```diff
@@ -85,7 +85,7 @@
 
 if not releaseExists:
     print('Creating release "%s" with tag "%s" on commit "%s"' % (title, tag, options.commit))
-    draft = True if options.tag else False
+    draft = False if tag.startswith('nightly_') else True
```
Comment (lukicdarkoo) [inline scripts/packaging/publish_release.py:88]: `tag` is a scoped variable, defined in if condition before. I am surprised it used to work.

Maybe we should define `tag` as a public variable, just `tag = None` somewhere in the beginning.

**Review by lukicdarkoo — CHANGES_REQUESTED** [review]
`tag` seems not to be declared as a public variable.

**Comment by omichel** [comment]
It seems to be slightly better like this, but the diff is bigger, so hopefully, it doesn't break anything...

**Comment by omichel** [comment]
When merged I will port it to the develop branch as well.

**Review by lukicdarkoo — APPROVED** [review]
Thank you

**Pull request merged**